### ISSUE TITLE
ONNX and Torch models

### ISSUE THREAD

**Issue by AlexandderGorodetski** [k2-fsa/icefall#1642]
Hello guys,

I have a question about ONNX and Torch models.

1. We can create ONNX models using export_onnx.py script in this case we get 3 models enoder.onnx, decoder.onnx and joiner.onnx
2.  We can create single Torch model using export.py script and in this case we get single model pretrained.pt.
3. For decoding using Torch model we use script pretrained.py.
4. For decoding using ONNX models we use script onnx_pretrained.py.
5. I found that transcription output of Torch and ONNX models are exactly same (I tested it on 100 utterances).
6. On other hand I found that encoder outputs of Torch and ONNX systems are completely different.
7. In addition logits values before argmax of Torch and ONNX systems are completely different, but argmax() values of both systems are exactly same!
8. My question is following. My product guys request from me to convert Encoder Torch model to ONNX format. But on other hand they prefer to use Decoder and Joiner in Torch format. 
9. I found that when I use ONNX encoder and Torch decoder and joiner I get illogical results.
10. I guess that I have somehow to convert ONNX decoder and joinder models to Torch format. Is such conversion possible?

Thanks,
AlexG.

**Comment by csukuangfj** [comment]
Note:

1. torch.encoder + torch.joiner.encoder_proj = onnx.encoder
2. torch.decoder + torch.joiner.decoder_proj = onnx.decoder
3. torch.joiner != onnx.joiner
4. torch.encoder != onnx.encoder
5. torch.decoder != onnx.decoder

6. The output of torch.joiner == the output of onnx.joiner if you give the same input.

So, you cannot compare the output of `torch.encoder` with `onnx.encoder` since the above `item 1`.

Please think it again and then everything should be clear to you.
